modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenResponse.java

```diff
@@ -93,7 +94,7 @@ public class CodegenResponse implements IJsonSchemaValidationProperties {
     @Override
     public int hashCode() {
         return Objects.hash(headers, code, message, examples, dataType, baseType, containerType, hasHeaders,
-                isString, isNumeric, isInteger, isLong, isNumber, isFloat, isDouble, isDecimal, isByteArray, isBoolean, isDate,
+                isString, isUnit, isNumeric, isInteger, isLong, isNumber, isFloat, isDouble, isDecimal, isByteArray, isBoolean, isDate,
```

Comment:
@chrisbartoloburlo thanks for the PR. I couldn't understand/follow what's `isUnit` or `unit`?

What about removing it from this PR so that other fixes can be merged?

Reply:
@wing328 It is being used to set the flag `unitReturnType` which is in turn used in `api.mustache` to coordinate the return type. `unit` is the return type in Scala that represents 'no value'. Perhaps we can find a more generic name. 

`isDefault` is being used in other places and I refrained from using

Reply:
Fixed in [0e65481](https://github.com/OpenAPITools/openapi-generator/pull/11949/commits/0e65481d7dd3ea1d6ab3e9ab14296a931a9e76cb)